use Video's own HAL mode for HLG at 60 fps

Smali changes of vendor_oplus_camera 337cce9, over baksmali --sequential-labels output.

diff --git a/classes/com/oplus/ocs/camera/producer/decision/OperationModeDecision.smali b/classes/com/oplus/ocs/camera/producer/decision/OperationModeDecision.smali
index 3b5393416ddfb803067fb598e5aaadbcf6668095..44bc4b3981148e495bd1e4cd03e9b45d9f6962f0 100644
--- a/classes/com/oplus/ocs/camera/producer/decision/OperationModeDecision.smali
+++ b/classes/com/oplus/ocs/camera/producer/decision/OperationModeDecision.smali
@@ -182,8 +182,8 @@
     return-object p0
 .end method
 
-.method public static halHlg(Lcom/oplus/ocs/camera/metadata/parameter/Parameter;Landroid/util/Size;)Z
-    .locals 2
+.method public static halHlg(Lcom/oplus/ocs/camera/metadata/parameter/Parameter;Landroid/util/Size;)I
+    .locals 3
 
     sget-object v0, Lcom/oplus/ocs/camera/metadata/UConfigureKeys;->VIDEO_DOLBY_MODE:Lcom/oplus/ocs/camera/metadata/ConfigureKey;
 
@@ -197,7 +197,13 @@
 
     move-result v0
 
-    if-eqz v0, :cond_28
+    if-eqz v0, :cond_38
+
+    if-eqz p1, :cond_38
+
+    invoke-virtual {p1}, Landroid/util/Size;->getWidth()I
+
+    move-result v2
 
     sget-object v0, Lcom/oplus/ocs/camera/metadata/UConfigureKeys;->VIDEO_FPS:Lcom/oplus/ocs/camera/metadata/ConfigureKey;
 
@@ -205,29 +211,42 @@
 
     move-result-object v0
 
-    const-string v1, "video_30fps"
+    const/16 v1, 0x780
 
-    invoke-virtual {v1, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+    const-string p0, "video_30fps"
 
-    move-result v0
+    invoke-virtual {p0, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
 
-    if-eqz v0, :cond_28
+    move-result p0
 
-    if-eqz p1, :cond_28
+    if-eqz p0, :cond_28
 
-    invoke-virtual {p1}, Landroid/util/Size;->getWidth()I
+    if-gt v2, v1, :cond_38
 
-    move-result v0
+    const/4 v0, 0x1
 
-    const/16 v1, 0x780
+    return v0
 
-    if-gt v0, v1, :cond_28
+    :cond_28
+    const-string p0, "video_60fps"
 
-    const/4 v0, 0x1
+    invoke-virtual {p0, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_38
+
+    if-lt v2, v1, :cond_38
+
+    const/16 v1, 0xf00
+
+    if-gt v2, v1, :cond_38
+
+    const/4 v0, 0x2
 
     return v0
 
-    :cond_28
+    :cond_38
     const/4 v0, 0x0
 
     return v0
@@ -795,19 +814,29 @@
 
     move-result-object v4
 
-    invoke-static {v4, v5}, Lcom/oplus/ocs/camera/producer/decision/OperationModeDecision;->halHlg(Lcom/oplus/ocs/camera/metadata/parameter/Parameter;Landroid/util/Size;)Z
+    invoke-static {v4, v5}, Lcom/oplus/ocs/camera/producer/decision/OperationModeDecision;->halHlg(Lcom/oplus/ocs/camera/metadata/parameter/Parameter;Landroid/util/Size;)I
 
     move-result v4
 
-    sput-boolean v4, Lcom/oplus/ocs/camera/producer/decision/OperationModeDecision;->sHalHlg:Z
+    if-eqz v4, :cond_24
+
+    const/4 v5, 0x1
+
+    sput-boolean v5, Lcom/oplus/ocs/camera/producer/decision/OperationModeDecision;->sHalHlg:Z
 
-    if-eqz v4, :cond_21
+    if-ne v4, v5, :cond_27
 
     const-string v4, "80a4"
 
     return-object v4
 
-    :cond_21
+    :cond_24
+    sput-boolean v4, Lcom/oplus/ocs/camera/producer/decision/OperationModeDecision;->sHalHlg:Z
+
+    goto :goto_27
+
+    :cond_27
+    :goto_27
     invoke-virtual {p0}, Lcom/oplus/ocs/camera/common/parameter/SdkCameraDeviceConfig;->getConfigureParameter()Lcom/oplus/ocs/camera/metadata/parameter/Parameter;
 
     move-result-object v0
@@ -818,14 +847,14 @@
 
     const/4 v1, 0x0
 
-    if-eqz v0, :cond_86
+    if-eqz v0, :cond_8c
 
-    if-nez p1, :cond_2f
+    if-nez p1, :cond_35
 
-    goto :goto_86
+    goto :goto_8c
 
     .line 139
-    :cond_2f
+    :cond_35
     invoke-static {}, Lcom/oplus/ocs/camera/producer/info/CameraConfigHelper;->getFeatureOperationMode()Ljava/util/Map;
 
     move-result-object v0
@@ -840,7 +869,7 @@
 
     const-string v2, "OperationModeDecision"
 
-    if-eqz v0, :cond_59
+    if-eqz v0, :cond_5f
 
     .line 140
     invoke-static {}, Lcom/oplus/ocs/camera/producer/info/CameraConfigHelper;->getFeatureOperationMode()Ljava/util/Map;
@@ -869,12 +898,12 @@
     invoke-static {v2, v0, v1}, Lcom/oplus/ocs/camera/common/util/CameraUnitLog;->d(Ljava/lang/String;Ljava/util/function/Function;Ljava/lang/Object;)V
 
     .line 145
-    :cond_59
+    :cond_5f
     invoke-static {v1, p0, p1}, Lcom/oplus/ocs/camera/producer/decision/OperationModeDecision;->parseFeatureOperationPriority(Ljava/util/Set;Lcom/oplus/ocs/camera/common/parameter/SdkCameraDeviceConfig;Ljava/util/Map;)Ljava/lang/String;
 
     move-result-object v0
 
-    if-nez v0, :cond_7d
+    if-nez v0, :cond_83
 
     .line 148
     invoke-virtual {p0}, Lcom/oplus/ocs/camera/common/parameter/SdkCameraDeviceConfig;->getConfigureParameter()Lcom/oplus/ocs/camera/metadata/parameter/Parameter;
@@ -889,12 +918,12 @@
 
     move-result-object v1
 
-    :cond_6b
+    :cond_71
     invoke-interface {v1}, Ljava/util/Iterator;->hasNext()Z
 
     move-result v3
 
-    if-eqz v3, :cond_7d
+    if-eqz v3, :cond_83
 
     invoke-interface {v1}, Ljava/util/Iterator;->next()Ljava/lang/Object;
 
@@ -907,10 +936,10 @@
 
     move-result-object v0
 
-    if-eqz v0, :cond_6b
+    if-eqz v0, :cond_71
 
     .line 157
-    :cond_7d
+    :cond_83
     new-instance p0, Lcom/oplus/ocs/camera/producer/decision/OperationModeDecision$$ExternalSyntheticLambda4;
 
     invoke-direct {p0}, Lcom/oplus/ocs/camera/producer/decision/OperationModeDecision$$ExternalSyntheticLambda4;-><init>()V
@@ -919,8 +948,8 @@
 
     return-object v0
 
-    :cond_86
-    :goto_86
+    :cond_8c
+    :goto_8c
     return-object v1
 .end method
 
